File: River/River.Body/Ahoy/Sockets/AhoySocket.py
import socket
from Ahoy.AhoyList import AhoyList

# in python endpoints are simply tuples (ipaddress,port)


class AhoySocket():
    def __init__(self, port, ipAddress="127.0.0.1"):

        self.__socket__ = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
        # self.__socket__.setblocking(True)
        # self.__socket__.
        # self.localEndPoint = ("0.0.0.0", port)
        self.localEndPoint = (ipAddress, port)
        self.__socket__.bind(self.localEndPoint)
        # self.remoteEndPoints = AhoyList()
        # self.__socket__.recv

    def __str__(self):
        return self.localEndPoint[0] + ":" + str(self.localEndPoint[1])

File: River/River.Droid/Assets/Packages/Ahoy.Net/Scripts/Sockets/AhoySocket.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Net.Sockets;
using System.Text;
// [Serializable]

using UnityEngine;
using Unity.Entities;
namespace Ahoy.Net
{

    [Serializable]
    public struct AhoySocket : ISharedComponentData, IEquatable<AhoySocket>
    {

        static int idIncr;
        public int id { get; private set; }

        public Socket socket;
        public IPEndPoint localEndPoint;
        public List<EndPoint> remoteEndPoints;

        public AhoySocket(string ipAddress, int port)
        {
            id = ++idIncr;
            remoteEndPoints = new List<EndPoint>();
            // localEndPoint = new IPEndPoint(IPAddress.Any, port);
            localEndPoint = new IPEndPoint(IPAddress.Parse(ipAddress), port);
            socket = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp);
            socket.Bind(localEndPoint);
            // socket.SendTimeout = timeOut;//these arent needed for datagrams i think
            // socket.ReceiveTimeout = timeOut;
        }

        public bool Equals(AhoySocket other)
        {
            return id == other.id;
        }

        public override int GetHashCode()
        {
            return id;
        }

        public override string ToString()
        {
            return $"{localEndPoint.Address}:{localEndPoint.Port}";
        }
    }
}